src/sentry/snuba/metrics.py

```diff
@@ -351,27 +401,29 @@ def _get_metric(metric_name: str) -> dict:
 
 
 class IndexMockingDataSource(DataSource):
-    def get_metrics(self, projects: Sequence[Project]) -> List[dict]:
+    def get_metrics(self, projects: Sequence[Project]) -> Sequence[MetricMeta]:
         """Get metrics metadata, without tags"""
         return [
-            dict(
+            MetricMeta(
                 name=name,
                 **{key: value for key, value in metric.items() if key != "tags"},
             )
             for name, metric in _METRICS.items()
         ]
 
-    def get_single_metric(self, projects: Sequence[Project], metric_name: str) -> dict:
+    def get_single_metric(
+        self, projects: Sequence[Project], metric_name: str
+    ) -> MetricMetaWithTagKeys:
         """Get metadata for a single metric, without tag values"""
         try:
             metric = _METRICS[metric_name]
         except KeyError:
             raise InvalidParams()
 
-        return dict(
+        return MetricMetaWithTagKeys(
```

Comment:
i don't think we ever instantiate types like that. is this necessary to get mypy's blessing?

Reply:
I think I made pyright complain a bit less about it. But this file is not fully typed yet anyway, so I've reverted this change.